internal/gensupport/resumable.go

```diff
@@ -241,13 +253,31 @@ func (rx *ResumableUpload) Upload(ctx context.Context) (resp *http.Response, err
 			default:
 			}
 
-			resp, err = rx.transferChunk(ctx)
+			var rCtx context.Context
+			var cancel context.CancelFunc
+
+			rCtx = ctx
+			if transferTimeout != 0 {
+				rCtx, cancel = context.WithTimeout(ctx, transferTimeout)
+				defer cancel()
```

Comment:
I'm not sure this is a big deal practically speaking, but I'm not sure `defer cancel()` is what you want here, since that doesn't return until the function returns. So we might have a bunch of contexts stack up over the life of this loop.

Maybe we can explicitly cancel after transferChunk returns instead?

Reply:
Good catch. added the cancel when context is done.